IOMC/EventVertexGenerators/src/BetafuncEvtVtxGenerator.cc

```diff
@@ -50,19 +47,20 @@ BetafuncEvtVtxGenerator::BetafuncEvtVtxGenerator(const edm::ParameterSet& p) : B
                                             << "Illegal resolution in Z (SigmaZ is negative)";
     }
   }
+  if (readDB_) {
+    beamToken_ = esConsumes<SimBeamSpotObjects, SimBeamSpotObjectsRcd, edm::Transition::BeginLuminosityBlock>();
+  }
 }
 
 BetafuncEvtVtxGenerator::~BetafuncEvtVtxGenerator() {}
 
 void BetafuncEvtVtxGenerator::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const& iEventSetup) {
   update(iEventSetup);
 }
-void BetafuncEvtVtxGenerator::beginRun(const edm::Run&, const edm::EventSetup& iEventSetup) { update(iEventSetup); }
```

Comment:
Getting from the EventSetup in both the BeginRun and BeginLuminosityBlock is unnecessary and would have required 2 different tokens.